src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -92,8 +92,9 @@ private sealed partial class UIThreadOnly : IDisposable
 
             private readonly IWaitIndicator _waitIndicator;
 
-            public readonly ITextBuffer OutputBuffer;
-            public readonly ITextBuffer StandardInputBuffer;
+            public ITextBuffer OutputBuffer { get; }
+            public ITextBuffer HistoryBuffer { get; private set; }
```

Comment:
`CurrentLanguageBuffer` refers to the input user just made, so `PrepareForInput` creates a new buffer every time SubmitAsync succeed. So there will be new buffers one way or another.

Can you tell me a bit more about language services problem?

---
In reply to: [62405034](https://github.com/dotnet/roslyn/pull/11078#discussion_r62405034) [](ancestors = 62405034,62404892)